WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -3056,59 +3021,23 @@ function WeakAuras.ReloadTriggerOptions(data)
-      displayOptions[id].args.trigger.args.type.set = options_set;
+      local aura_options = WeakAuras.GetBuffTriggerOptions(data);
+      trigger_options = union(trigger_options, aura_options);
     elseif(optionTriggerChoices[id] >= 1 and (getAll(data, {"trigger", "type"}) == "event" or getAll(data, {"trigger", "type"}) == "status")) then
       local event = getAll(data, {"trigger", "event"});
       local unevent = getAll(data, {"trigger", "unevent"});
       if(event and WeakAuras.event_prototypes[event]) then
-        local trigger_options_created;
-        if(event == "Combat Log") then
-          local subeventPrefix = getAll(data, {"trigger", "subeventPrefix"});
-          local subeventSuffix = getAll(data, {"trigger", "subeventSuffix"});
-          if(subeventPrefix and subeventSuffix) then
-            trigger_options_created = true;
-            displayOptions[id].args.trigger.args = union(trigger_options, WeakAuras.ConstructOptions(WeakAuras.event_prototypes[event], data, 10, subeventPrefix, subeventSuffix, optionTriggerChoices[id], nil, unevent));
-          end
-        end
-
-        if not(trigger_options_created) then
-          displayOptions[id].args.trigger.args = union(trigger_options, WeakAuras.ConstructOptions(WeakAuras.event_prototypes[event], data, 10, nil, nil, optionTriggerChoices[id], nil, unevent));
-        end
-      else
-        displayOptions[id].args.trigger.args = union(trigger_options, {});
-        removeFuncs(displayOptions[id].args.trigger);
-      end
-      removeFuncs(displayOptions[id].args.trigger);
-      replaceNameDescFuncs(displayOptions[id].args.trigger, data);
-      replaceImageFuncs(displayOptions[id].args.trigger, data);
-      replaceValuesFuncs(displayOptions[id].args.trigger, data);
-
-      if(displayOptions[id].args.trigger.args.unevent) then
-        displayOptions[id].args.trigger.args.unevent.set = options_set;
-      end
-      if(displayOptions[id].args.trigger.args.subeventPrefix) then
-        displayOptions[id].args.trigger.args.subeventPrefix.set = options_set
-      end
-      if(displayOptions[id].args.trigger.args.subeventSuffix) then
-        displayOptions[id].args.trigger.args.subeventSuffix.set = options_set;
-      end
-
-      if(displayOptions[id].args.trigger.args.type) then
-        displayOptions[id].args.trigger.args.type.set = options_set;
+        trigger_options = union(trigger_options, WeakAuras.ConstructOptions(WeakAuras.event_prototypes[event], data, 10, optionTriggerChoices[id], nil, unevent));
       end
-      if(displayOptions[id].args.trigger.args.event) then
-        displayOptions[id].args.trigger.args.event.set = options_set;
-      end
-    else
-      displayOptions[id].args.trigger.args = trigger_options;
-      removeFuncs(displayOptions[id].args.trigger);
     end
+    displayOptions[id].args.trigger.args = trigger_options;
+
+    removeFuncs(displayOptions[id]);
+    replaceNameDescFuncs(displayOptions[id], data);
```

Comment:
Those 3 calls, merely moved. The code between the old and new place does not affect those functions, and keeping them all together makes the code easier to understand